### PR TITLE
feat(TCOMP-1814): Standalone connectors support

### PR DESCRIPTION
**Pull request by chmyga**
### Requirements

* Any code change adding any logic **MUST** be tested through a unit test executed with the default build
* Any API addition **MUST** be done with a documentation update if relevant 

### Why this PR is needed?
https://jira.talendforge.org/browse/TCOMP-1814

### What does this PR adds (design/code thoughts)?
Standalone connectors support and processing in framework. Does not include beam/cloud implementation and processing.

### CODE REVIEW

**Review by undx — COMMENTED** [review]
Apparently, LGTM! Will try to do some tests this week-end for checking, will approve afterwards.

Great task Dmytro ! 👍🏻

**Review by undx — APPROVED** [review]
LGTM 👍🏻

**Review by ypiel-talend — COMMENTED** [review]
LGTM

File: component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java [hunk 1]
```diff
@@ -1903,6 +1932,46 @@ public String name() {
...
+                            .supports(org.talend.sdk.component.runtime.standalone.DriverRunner.class)
+                                    ? config -> executeInContainer(plugin,
+                                            () -> context
+                                                    .getOwningExtension()
+                                                    .convert(new ComponentInstanceImpl(doInvoke(
+                                                            constructor, parameterFactory.apply(config)), plugin,
+                                                            component.getName(), name),
+                                                            org.talend.sdk.component.runtime.standalone.DriverRunner.class))
+                                    : config -> new DriverRunnerImpl(this.component.getName(), name, plugin,
+                                            doInvoke(constructor, parameterFactory.apply(config)));
```
Comment (clesaec) [inline component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java:1959]: Would be easier to understand with code like : 
```java
final boolean isDriverSupport = context.getOwningExtension() != null && context
                            .getOwningExtension()
                            .supports(org.talend.sdk.component.runtime.standalone.DriverRunner.class);

final Function<Map<String, String>, org.talend.sdk.component.runtime.standalone.DriverRunner> instantiator =
    (Map<String, String> config) -> {
            final Serializable instance = doInvoke(constructor, parameterFactory.apply(config));
            if (isDriverSupport) {
                     final ComponentInstanceImpl cpnInstance = new ComponentInstanceImpl(instance, plugin,
                                                            component.getName(), name);
                     return executeInContainer(plugin,
                               context.getOwningExtension()
                                           .convert(cpnInstance,
                                                            org.talend.sdk.component.runtime.standalone.DriverRunner.class))
              }
              else {
                       new DriverRunnerImpl(this.component.getName(), name, plugin,   instance);
            }
   }
```
but that's my poor brain which is not adapt enough :)

**Review by clesaec — APPROVED** [review]
:+1: 
with one comment

**Pull request merged**